### ISSUE TITLE
Add infrared jamming attribute

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#3323]
I can imagine a few ways a ship might make itself _look_ cooler to missiles without actually generating less overall heat:
- Low-E exterior hull plating
- Interior hull insulation
- Internal heat storage systems

(And I would like to add some pirate outfits that do this)

Looking at the code in `Projectile.cpp`, it seems like this should be easy conceptually, but I'm hesitant to offer up a patch because anything I change to add IR jamming to would alter the lock probability for IR-guided weapons.

**Comment by endless-sky** [comment]
The point of infrared tracking is to add an incentive to design cool-running ships, and to make it so the three tracking types are very different from each other and have to be defeated in different ways. Making it so you can "jam" infrared tracking would reduce that distinctiveness.

**Comment by KiLEdEnNis** [comment]
how about adding the possibility but just use it for certain aliens? (for example I'd love an "infrared jamming" for my Derogam as they have a high built-in heat, so they're usually relatively high in heat)

**Comment by accursedCursive** [comment]
The real problem is that jamming is necessary because missiles have their tracking quality set way too high, to the point that they never fail to track a target that doesn't have systems in place to interfere with tracking (if possible).
Meteor missiles use infrared tracking, so there should be some targets that they'll fail to track, but I have yet to fire a missile with a target selected and not have it point to something.
Sidewinder missiles have near-max radar tracking, so unless you devote 14 outfit space just to jam them (which is completely impractical for small ships, the only thing that Sidewinder threatens), they'll track you pretty consistently.
Torpedo missiles have optical tracking, which means they should struggle with small ships, but I can fire torpedos at a distant interceptor.

This is a balancing issue more than anything else, you need to find tracking values that make it a viable option to have missiles ignore you instead of just dodging them or shooting them.
Additionally, missiles should use tracking methods that make sense in terms of game balance.
Torpedos use optical tracking, which is good: They track large ships better, and can't easily swat small ships. Except that their optical tracking is way too good, they can track a Sparrow from far away.
Sidewinders use radar tracking, which is questionable: The only ships that can viably use radar jammers are the ones that Sidewinders can barely harm.
Meteors use thermal tracking, which is bad: They can destroy hot-running interceptors and fighters from afar in a matter of seconds but tend to ignore cruisers, even though everything else about their design and implementation suggests they're for allowing small ships to bomb large ships.

I suggest that Sidewinder has 0.7 thermal tracking, Meteor has radar tracking (Not sure how much, haven't experimented much with jammers), Torpedo gets dropped to 0.5 optical tracking, and Typhoon gets dropped to 0.6.

**Comment by Knightroad** [comment]
I like this idea, and if used with dropping the tracking ability of the missiles, it would make the jammers worthwhile. As it stands, it's more practical to have one or two anti-missile systems per ship.

**Comment by Zitchas** [comment]
Some random ship in game broadcast a message about "Keep your ship cool to avoid infrared missiles." As a result I've outfitted my ship to run ice-cold under everything but intense combat. I think I might be overdoing it, but I'm still learning the balance. Anyway, the point is that despite keeping my heat rating pretty much locked to zero, I don't think I've ever seen a meteor missile pointed my way that failed to hit me (unless I managed to outrun it). So while I can't comment on the other merits of things, toning down the tracking on the meteors a bit so that staying cold actually does work would be nice.

As a side note, EV had flares, which were a cool concept and one that made a lot of sense in terms of dealing with heat seeking missiles. How would it affect game balance to use flares for decoying infrared missiles instead of having something to jam them?

**Comment by jafdy** [comment]
The missiles that loose lock are the ones that don't come anywhere near you, so you don't see them anyway.

**Comment by Zitchas** [comment]
Ah! So the lock / no lock is determined at launch, and doesn't change thereafter?

**Comment by tehhowch** [comment]
Lock status is checked multiple times during flight. When a missile loses its lock it will simply fly straight. So, if you are also traveling in a straight line, i.e. trying to outrun it by moving either towards or away from it, you will not notice that it is not currently tracking you.

**Comment by Pointedstick** [comment]
> Lock status is checked multiple times during flight. When a missile loses its lock it will simply fly straight. So, if you are also traveling in a straight line, i.e. trying to outrun it by moving either towards or away from it, you will not notice that it is not currently tracking you.

Which would (and IMO should) be addressed by https://github.com/endless-sky/endless-sky/issues/1865

**Comment by endless-sky** [comment]
Of course, if someone is trying to dodge a missile by flying directly away from it, they clearly don't understand what "dodging" means...

**Comment by DingusShingleton** [comment]
Due to the physics of the game, ie the fact that conservation of momentum isn't a thing, it sometimes is simply in your best interests to fly in a straight line. If your speed falls into a certain range, as soon as you thrust in a different direction you slow down and immediately get plastered with missiles. 

NB I'm not saying there's a problem with the physics, only that they sometimes  encourage the "straight line dodge" maneuver.
